File: backend/models/ResumeReport.js
const mongoose = require('mongoose');

const resumeReportSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true
  },
  resumeId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Resume',
    default: null
  },
  fileName: {
    type: String,
    default: ''
  },
  // Extracted content
  skills: {
    type: Array,
    default: []
  },
  missingSkills: {
    type: Array,
    default: []
  },
  projects: {
    type: Array,
    default: []
  },
  education: {
    type: Array,
    default: []
  },
  experience: {
    type: Array,
    default: []
  },
  certifications: {
    type: Array,
    default: []
  },
  // AI evaluation
  strengths: {
    type: Array,
    default: []
  },
  weaknesses: {
    type: Array,
    default: []
  },
  atsScore: {
    type: Number,
    default: 0
  },
  resumeScore: {
    type: Number,  // 0 – 100
    default: 0
  },
  recommendedRole: {
    type: String,
    default: ''
  },
  improvementSuggestions: {
    type: Array,
    default: []
  },
  overallFeedback: {
    type: String,
    default: ''
  },
  createdByAI: {
    type: Boolean,
    default: true
  }
}, { timestamps: true });

module.exports = mongoose.model(
  'ResumeReport',
  resumeReportSchema,
  'ResumeReports'
);

File: backend/models/resume_report.py
"""models/resume_report.py — ResumeReport table (replaces Mongoose ResumeReport.js)"""
from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime, ForeignKey, Text, JSON
from sqlalchemy.sql import func
from database import Base


class ResumeReport(Base):
    __tablename__ = "resume_reports"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    resume_id = Column(Integer, ForeignKey("resumes.id"), nullable=True)
    file_name = Column(String(500), default="")
    skills = Column(JSON, default=list)
    missing_skills = Column(JSON, default=list)
    projects = Column(JSON, default=list)
    education = Column(JSON, default=list)
    experience = Column(JSON, default=list)
    certifications = Column(JSON, default=list)
    strengths = Column(JSON, default=list)
    weaknesses = Column(JSON, default=list)
    ats_score = Column(Float, default=0)
    resume_score = Column(Float, default=0)
    recommended_role = Column(String(255), default="")
    improvement_suggestions = Column(JSON, default=list)
    overall_feedback = Column(Text, default="")
    created_by_ai = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
